tests/e2e/workflows/workflow_test.go

```diff
@@ -155,132 +168,101 @@ func raiseEventTest(url string, instanceID string) string {
 	}
 
 	// Purge the workflow
 	postString = fmt.Sprintf("%s/PurgeWorkflow/dapr/%s", url, instanceID)
 	resp, err = utils.HTTPPost(postString, nil)
 	if err != nil {
-		return fmt.Sprintf("Failure purging workflow: %s", err.Error())
+		return fmt.Errorf("failure purging workflow: %w", err)
 	}
 
 	// Startup a new workflow with the same instanceID to ensure that it is available
 	postString = fmt.Sprintf("%s/StartWorkflow/dapr/placeOrder/%s", url, instanceID)
 	resp, err = utils.HTTPPost(postString, nil)
 	if err != nil {
-		return fmt.Sprintf("Failure starting workflow: %s", err.Error())
+		return fmt.Errorf("failure starting workflow: %w", err)
 	}
 
 	instanceID = string(resp)
 	resp, err = utils.HTTPGet(getString)
 	if err != nil {
-		return fmt.Sprintf("Failure getting info on newly started workflow: %s", err.Error())
+		return fmt.Errorf("failure getting info on newly started workflow: %w", err)
 	}
 	if string(resp) != "Running" {
-		return fmt.Sprintf("Expected workflow to be Running, actual workflow state is: %s", string(resp))
+		return fmt.Errorf("expected workflow to be Running, actual workflow state is: %s", string(resp))
 	}
 
-	return string(resp)
-}
-
-var workflowAppTests = []struct {
-	in               string
-	instanceID       string
-	app              string
-	expectedResponse string
-}{
-	{
-		"start",
-		"startID",
-		"workflowsapp",
-		"Running",
-	},
-
-	{
-		"pauseResume",
-		"pauseID",
-		"workflowsapp",
-		"Success",
-	},
-
-	{
-		"purge",
-		"purgeID",
-		"workflowsapp",
-		"Running",
-	},
-
-	{
-		"raiseEvent",
-		"raiseEventID",
-		"workflowsapp",
-		"Completed",
-	},
+	return nil
 }
 
 func TestWorkflow(t *testing.T) {
-	for _, tt := range workflowAppTests {
-		t.Run(tt.in, func(t *testing.T) {
-			// Get the ingress external url of test app
-			externalURL := tr.Platform.AcquireAppExternalURL(tt.app)
-			require.NotEmpty(t, externalURL, "external URL must not be empty")
-
-			// Check if test app endpoint is available
-			_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
-			require.NoError(t, err)
-
-			result := "false"
-			switch tt.in {
-			case "purge":
-				result = purgeTest(externalURL, tt.instanceID)
-			case "start":
-				result = startTest(externalURL, tt.instanceID)
-			case "pauseResume":
-				result = pauseResumeTest(externalURL, tt.instanceID)
-			case "raiseEvent":
-				result = raiseEventTest(externalURL, tt.instanceID)
-			}
-
-			require.Equal(t, tt.expectedResponse, result)
-		})
-	}
+	// Get the ingress external url of test app
+	externalURL := getAppEndpoint()
+	require.NotEmpty(t, externalURL, "external URL must not be empty")
+
+	// Check if test app endpoint is available
+	require.NoError(t, utils.HealthCheckApps(externalURL))
+
+	// Generate a unique test suffix for this test
+	suffixBytes := make([]byte, 7)
+	_, err := io.ReadFull(rand.Reader, suffixBytes)
+	require.NoError(t, err)
+	suffix := hex.EncodeToString(suffixBytes)
+
+	// Run tests
+	t.Run("Start", func(t *testing.T) {
```

Comment:
One thing I failed to realize with the previous Run structure of this test was that when a failure occurred inside the function/test, we didn't get the line number to trace exactly where the failure was. With this new method of passing in the "testing" pointer, will this provide the line number in the instance of failure? If not, how difficult would it be to add that? (definitely not a blocker for this review)

Reply:
This will help a bit because at least you can see the line that failed in the test (like L254). But you won't see the exact line that failed in the test function, because that doesn't have the `t` and doesn't do assertions. We can change that if needed (maybe a separate PR)